pkg/sippyserver/server.go

```diff
@@ -533,6 +537,81 @@ func (s *Server) jsonTestOutputsFromDB(w http.ResponseWriter, req *http.Request)
 	api.RespondWithJSON(http.StatusOK, w, outputs)
 }
 
+func (s *Server) jsonComponentsFromBigQuery(w http.ResponseWriter, req *http.Request) {
+	baseRelease := req.URL.Query().Get("baseRelease")
+	sampleRelease := req.URL.Query().Get("sampleRelease")
+	if baseRelease == "" ||
+		sampleRelease == "" {
+		api.RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{
+			"code":    http.StatusBadRequest,
+			"message": "missing required parameters.",
+		})
+		return
+	}
+
+	timeStr := req.URL.Query().Get("baseStartTime")
+	baseStartTime, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		api.RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{
+			"code":    http.StatusBadRequest,
+			"message": "base start time in wrong format.",
+		})
+		return
+	}
+	timeStr = req.URL.Query().Get("baseEndTime")
+	baseEndTime, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		api.RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{
+			"code":    http.StatusBadRequest,
+			"message": "base end time in wrong format.",
+		})
+		return
+	}
+	timeStr = req.URL.Query().Get("sampleStartTime")
+	sampleStartTime, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		api.RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{
+			"code":    http.StatusBadRequest,
+			"message": "sample start time in wrong format.",
+		})
+		return
+	}
+	timeStr = req.URL.Query().Get("sampleEndTime")
+	sampleEndTime, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		api.RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{
+			"code":    http.StatusBadRequest,
+			"message": "sample end time in wrong format.",
+		})
+		return
+	}
+
+	component := req.URL.Query().Get("component")
+	capability := req.URL.Query().Get("capability")
+	testId := req.URL.Query().Get("test_id")
+
+	platform := req.URL.Query().Get("platform")
+	upgrade := req.URL.Query().Get("upgrade")
+	arch := req.URL.Query().Get("arch")
+	network := req.URL.Query().Get("network")
+
+	groupBy := req.URL.Query().Get("group_by")
+
+	outputs, errs := api.GetComponentReportFromBigQuery(s.bigQueryClient, baseRelease, sampleRelease, component, capability, platform, upgrade, arch, network, testId, groupBy, baseStartTime, baseEndTime, sampleStartTime, sampleEndTime)
+	if len(errs) > 0 {
+		log.Warningf("%d errors were encountered while querying component from big query:", len(errs))
+		for _, err := range errs {
+			log.Error(err.Error())
+		}
+		api.RespondWithJSON(http.StatusInternalServerError, w, map[string]interface{}{
+			"code":    http.StatusInternalServerError,
+			"message": "error querying component from big query",
```

Comment:
this would've save me a lot of time debugging because I would've seen it in the request payload:

```
WARN[2023-05-01T07:56:08.25-04:00] 2 errors were encountered while querying component from big query: 
ERRO[2023-05-01T07:56:08.25-04:00] googleapi: Error 400: Unrecognized name: variant; Did you mean variants? at [10:250], invalidQuery 
ERRO[2023-05-01T07:56:08.25-04:00] googleapi: Error 400: Unrecognized name: variant; Did you mean variants? at [10:250], invalidQuery
```